File: backend/mallapi/src/test/java/org/nareun130/mallapi/service/ProductServiceTests.java
package org.nareun130.mallapi.service;

import java.util.List;
import java.util.UUID;

import org.junit.jupiter.api.Test;
import org.nareun130.mallapi.dto.PageRequestDTO;
import org.nareun130.mallapi.dto.PageResponseDTO;
import org.nareun130.mallapi.dto.ProductDTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import lombok.extern.log4j.Log4j2;

@SpringBootTest
@Log4j2
public class ProductServiceTests {
    
    @Autowired
    ProductService productService;

    @Test
    public void testList() {

        PageRequestDTO pageRequestDTO = PageRequestDTO.builder().build();

        PageResponseDTO<ProductDTO> result = productService.getList(pageRequestDTO);

        result.getDtoList().forEach(dto -> log.info(dto));
    }

    @Test
    public void testRegister() {

        ProductDTO productDTO = ProductDTO.builder()    
            .pname("새로운 상품")
            .pdesc("신규 추가 상품")
            .price(1000)
            .build();

        //! uuid 필요!
        productDTO.setUploadFileNames(
            List.of(
                UUID.randomUUID()+"_" + "TEST1.jpg",
                UUID.randomUUID()+"_" + "TEST2.jpg"));
        
        productService.register(productDTO);
    }
    

    
}

File: backend/mallapi-kotlin/src/test/kotlin/com/nareun/mallapikotlin/service/ProductServiceTests.kt
package com.nareun.mallapikotlin.service

import com.nareun.mallapikotlin.dto.PageRequestDTO
import com.nareun.mallapikotlin.dto.PageResponseDTO
import com.nareun.mallapikotlin.dto.ProductDTO
import lombok.extern.log4j.Log4j2
import org.junit.jupiter.api.Test
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import java.util.*

@SpringBootTest
@Log4j2
class ProductServiceTests {
    @Autowired
    private lateinit var productService : ProductService

    private val log = LoggerFactory.getLogger(ProductServiceTests::class.java)
    @Test
    fun testList(){
        val pageRequestDTO : PageRequestDTO = PageRequestDTO()

        val result : PageResponseDTO<ProductDTO> = productService.getList(pageRequestDTO)

        result.dtoList.forEach { dto ->  log.info(dto.toString())}

    }

    @Test
    fun testRegister(){
        val productDTO : ProductDTO = ProductDTO(
            pname = "새로운 상품",
            pdesc = "신규 추가 상품",
            price = 1000
        )
        productDTO.uploadFileNames = listOf(
            "${UUID.randomUUID()}_Test1.jpg",
            "${UUID.randomUUID()}_Test2.jpg",
        )

        productService.register(productDTO)
    }


}